pkg/payload/render.go

```diff
@@ -31,6 +35,35 @@ func Render(outputDir, releaseImage, clusterProfile string) error {
 		}
 	)
 
+	var requiredFeatureSet *string
+	if featureGateManifestPath != "" {
+		manifests, err := manifest.ManifestsFromFiles([]string{featureGateManifestPath})
+		if err != nil {
+			return fmt.Errorf("loading FeatureGate manifest: %w", err)
+		}
+		if len(manifests) != 1 {
+			return fmt.Errorf("FeatureGate manifest %s contains %d manifests, but expected only one", featureGateManifestPath, len(manifests))
+		}
+		featureGateManifest := manifests[0]
+		expectedGVK := schema.GroupVersionKind{Kind: "FeatureGate", Version: configv1.GroupVersion.Version, Group: config.GroupName}
+		if featureGateManifest.GVK != expectedGVK {
+			return fmt.Errorf("FeatureGate manifest %s GroupVersionKind %v, but expected %v", featureGateManifest.OriginalFilename, featureGateManifest.GVK, expectedGVK)
+		}
+		featureSet, found, err := unstructured.NestedString(featureGateManifest.Obj.Object, "spec", "featureSet")
+		if err != nil {
+			return fmt.Errorf("%s spec.featureSet lookup was not a string: %w", featureGateManifest.String(), err)
+		} else if found {
+			requiredFeatureSet = &featureSet
+			klog.Infof("--feature-gate-manifest-path=%s results in feature set %q", featureGateManifest.OriginalFilename, featureSet)
+		} else {
+			requiredFeatureSet = ptr.To[string]("")
```

Comment:
```suggestion
		
```

Reply:
deferring for now, in the rush to hit 4.17 branching.